python/lib/monitor/dmod/monitor/que_monitor.py

```diff
@@ -318,6 +309,79 @@ def resubmit(self, service_dict: dict, service_name: str) -> docker.from_env().s
+                    if mpirun in service_log_str:
+                        string = service_log_str
+                        word = string.split()
+                        print("word is instance of list:", isinstance(word, list))
+                        print("exitcode = ", word[-1])
+                        exitcode = int(word[-1])
+                        print("int exitcode = ", exitcode)
+                        if exitcode > 128 and exitcode < 256:
+                            fatal_signal_code = exitcode - 128
+                            print("service name: {}, fatal signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 4:
+                                print("service name: {}, Illegal Instruction, signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 6:
+                                print("service name: {}, Abort Signal, signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 8:
+                                print("service name: {}, Floating Point Exception, signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 9:
+                                print("service name: {}, Kill Signal, signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 11:
+                                print("service name: {}, Invalid Memory Reference, signal code: {}".format(service.name, fatal_signal_code))
+                            if fatal_signal_code == 13:
+                                print("service name: {}, Broken Pipe, signal code: {}".format(service.name, fatal_signal_code))
+                        if exitcode == 0:
+                            print("service name: {}, exit code: {}, Successful Completion".format(service.name, exitcode))
+                        if exitcode == 1:
+                            print("service name: {}, exit code: {}, Execution failed".format(service.name, exitcode))
+                        if exitcode == 2:
+                            print("service name: {}, exit code: {}, Misuse of shell builtins".format(service.name, exitcode))
+                        if exitcode == 126:
+                            print("service name: {}, exit code: {}, Command invoked cannot execute".format(service.name, exitcode))
+                        if exitcode == 127:
+                            print("service name: {}, exit code: {}, Command not found".format(service.name, exitcode))
+                        if exitcode == 130:
+                            print("service name: {}, exit code: {}, Script terminated by Control-C".format(service.name, exitcode))
+                        if exitcode == 139:
+                            print("service name: {}, exit code: {}, Segmentation Fault".format(service.name, exitcode))
+        
```

Comment:
Might be nice to put these error strings in a map so we have `error_strings = { 4, 'service name: {}, Illegal Instruction, signal code: {}'}` then this large switch could be replaced by `error_strings.get(exitcode, default).format(service.name, exitcode)` and we can provide a default error string